test/value.cpp

```diff
@@ -1589,7 +1589,7 @@ class value_test
         BOOST_TEST(
             get<1>(v2).get_string() == "value");
 
-#if __cplusplus >= 201703L
+#if __cplusplus >= 201703L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
         auto const [kc, vc] = v1;
```

Comment:
does `vc` receive a copy of the `json::value`?

Reply:
Correct; is not a desired behavior?

Reply:
I haven't used structured bindings, so I don't know. What if you want a reference? do you write  `auto const& [kc, kv] = v1`? Or do you write `auto const [kc, &kv] = v1]` ?

Reply:
`auto const& [kc, kv]`
See the various `static_assert`s in the added test code
I'm still struggling to get rvalues to work tho; I've actually fixed Richard's second comment, but didn't close it yet because I now want `auto [k, v] = kvp_prvalue` to be well formed and move the value, and likewise with `auto&& [krr, vrr] = kvp_prvalue`
Been staring at [\[dcl.struct.bind\]#4](https://eel.is/c++draft/dcl.struct.bind#4) and I'm still unsure as to why I can't both lvalues and rvalues to work without copying when `auto` and `auto&&` are used.

Reply:
`auto const& [kc, kv]` is interesting though, do you get a reference to a temporary `string_view`?

Reply:
No; the `string_view` is copied
```cpp
auto const& [kcr, vcr] = v1;
static_assert(std::is_same_v<decltype(kcr), string_view const>);
static_assert(std::is_same_v<decltype(vcr), value const&>);
```